docs.md

```diff
@@ -1,23 +1,396 @@
+   The architecture of the API is based on the following principles:
+   - The API is written in Python and FastAPI framework.
+   - The API is written in a way that it is easy to use, extend, test and maintain.
 
 ![Screenshot 2022-06-13 at 09 55 43](https://user-images.githubusercontent.com/67229938/173343113-d51d72b4-84c8-4c3b-8555-af41e59cd2de.png)
 
-### Detailed explanation of commands with examples: what they are and how to use them.
-
-#### 1. input_props_equals
-This command has different logic to handle series of equality checks.
-- Handling '*' as the wildcard flag
-This logic handles all the paths after a particular section. if /collections is supplied as the option, all the routes after it will be allowed e.g allow /collections/obs, allow /collections/test-data/obs, allow /collections/obs.
-`allow {
-   path = /collections/...
-}`
-If a particular path parameter is to be exempted, the command matches all other parameters aside the exempted one. e.g allow /collections/obs, allow /collections/test-data/obs, allow /collections/obs. deny /collections/lakes
-`allow {
-   path = /collections/...
-}`
-`deny {
-   path = /collections/lakes
-}`
--  Handling equality check between a property on the request object and a parameter
\ No newline at end of file
+View the Architecture in the browser [Here](https://www.figma.com/file/684S7kO4dPQbbZFZr6xZOn/Rego-builder?node-id=0%3A1) 
+
+
+
+   Installation
+   ============
+- Manual installation - Available [Here](https://github.com/r-scheele/rego_builder#readme)
+
+- Build your own image and start the container -
+   ```bash
+   $ Configure environment variables - the .env file
+   $ docker-compose -f docker-compose.dev.yml up -d
+
+   ```
+- Pull the official image from Docker Hub - and start the container 
+
+   ```bash
+   $ docker pull rscheele3214/rego-builder:latest
+   $ docker run rscheele3214/rego-builder:latest
+   ```
+
+
+How the translation from JSON to REGO works?
+============
+The REGO policies are created by a set of functions that serves as the translation logic, They're called commands. Each of these commands is responsible for writing specific REGO rule to a `.rego` file. However, the JSON to REGO conversion must follow certain conditions as defined by a pydantic model for effectiveness. The following is an example of how a rule is defined:
+
+   ```json     
+   {
+      "command": "input_prop_equals",
+      "properties": {
+         "input_property": "request_method",
+         "value": "GET"
+      }
+   }
+   ```
+In this case, The command key represents the operation to be performed and the properties key represents the properties that are being used in the operation.
+`input_prop_equals` is the command in example, which initiates the appropriate operation, on the properties object.
+The above rule translates to an equality check between the input_property and the value. <br />
+The REGO equivalent of the above rule object is: <br />
+   `input.request_method == "GET"`
+
```

Comment:
You should start this section with a lead like this:

A rule is defined by two keys: command and properties. The command key holds one of the recognized commands and the properties hold another dictionary containing the `input_property` and `value`. In special cases, the `datasource_items` items are also used/added etc etc.